woosmapgeofencing/src/main/java/com/webgeoservices/woosmapgeofencing/SearchAPIDataModel/SearchAPIResponseItem.java

```
@@ -0,0 +1,270 @@
+                ));
+
+                detailsResponseItem.geometry = geometryDetail;
+            }
+
+        }catch (Exception ex){
+            Log.e(TAG,ex.getMessage());
+        }
+        return detailsResponseItem;
+    }
+
+
+    /***
+     * Creates and returns <code>userPropertiesFiltered</code> object from the user_properties in the response Search API
+     * @param jsonObject RAW json response returned by <code>store</code> API
+     * @return <code>userPropertiesFiltered</code> object
+     */
+    public static HashMap<String, Object> getUserProperties(JSONObject jsonObject, String storeId){
+        HashMap<String, Object> userPropertiesFiltered = new HashMap<String, Object>();
+        SearchAPIResponseItem detailsResponseItem=new SearchAPIResponseItem();
+        JSONObject properties;
+
+        try{
+            JSONArray features = jsonObject.getJSONArray("features");
+            if (features.length() > 0) {
+                for(int i=0;i<features.length();i++){
+                    JSONObject feature = features.getJSONObject( i );
+                    properties=feature.getJSONObject("properties");
+                    if(properties.getString("store_id").equals(storeId)) {
+                        detailsResponseItem.userProperties = new Gson().fromJson(properties.get("user_properties").toString(), HashMap.class);
+                        if(WoosmapSettings.userPropertiesFilter.isEmpty()){
+                            userPropertiesFiltered = detailsResponseItem.userProperties;
+                        } else {
+                            for (String key : WoosmapSettings.userPropertiesFilter) {
+                                userPropertiesFiltered.put( key, detailsResponseItem.userProperties.get( key ) );
+                            }
+                        }
+                    }
+                }
+            }
+        }catch (Exception ex){
+            Log.e(TAG,ex.getMessage());
```

N'y-a-t-il pas de risque de fail silently comme au-dessus ? (et de crash ensuite)